tests/queries/0_stateless/02292_hash_array_tuples.reference

```diff
@@ -1,5 +1,19 @@
+arrays
 14617701568871014978
 12913842429399915005
 8351543757058688770
 12732328028874882204
 12371801021764949421	Array(Tuple(UInt8, Array(Tuple(UInt8, Tuple(UInt8, UInt8, Array(Tuple(UInt8, UInt8)))))))
+14617701568871014978
+12913842429399915005
+8351543757058688770
+12732328028874882204
+maps
+14617701568871014978
```

Comment:
Not sure if should map and array that represent the same data have the same hash. I see pros and cons for both variants. For the number of different types, we would get the same value until it's not some wide integer:

```sql
:) SELECT cityHash64(42 :: UInt8), cityHash64(42 :: UInt64), cityHash64(42 :: Int64), cityHash64(42 :: UInt128) FORMAT Vertical

Row 1:
──────
cityHash64(CAST('42', 'UInt8')):   11490350930367293593
cityHash64(CAST('42', 'UInt64')):  11490350930367293593
cityHash64(CAST('42', 'Int64')):   11490350930367293593
cityHash64(CAST('42', 'UInt128')): 16297183448592043045
```

Maybe let's add a few words about behavior into docs?